bundle/config/validate/folder_permissions.go

```diff
@@ -0,0 +1,154 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/permissions"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/databricks-sdk-go/service/workspace"
+	"golang.org/x/sync/errgroup"
+)
+
+type folderPermissions struct {
+}
+
+// Apply implements bundle.ReadOnlyMutator.
+func (f *folderPermissions) Apply(ctx context.Context, b bundle.ReadOnlyBundle) diag.Diagnostics {
+	if len(b.Config().Permissions) == 0 {
+		return nil
+	}
+
+	paths := []string{b.Config().Workspace.RootPath}
+
+	if !strings.HasPrefix(b.Config().Workspace.ArtifactPath, b.Config().Workspace.RootPath) {
+		paths = append(paths, b.Config().Workspace.ArtifactPath)
+	}
+
+	if !strings.HasPrefix(b.Config().Workspace.FilePath, b.Config().Workspace.RootPath) {
+		paths = append(paths, b.Config().Workspace.FilePath)
+	}
+
+	if !strings.HasPrefix(b.Config().Workspace.StatePath, b.Config().Workspace.RootPath) {
+		paths = append(paths, b.Config().Workspace.StatePath)
+	}
+
+	if !strings.HasPrefix(b.Config().Workspace.ResourcePath, b.Config().Workspace.RootPath) {
+		paths = append(paths, b.Config().Workspace.ResourcePath)
+	}
+
+	var diags diag.Diagnostics
+	errGrp, errCtx := errgroup.WithContext(ctx)
+	for _, path := range paths {
+		p := path
+		errGrp.Go(func() error {
+			diags = diags.Extend(checkFolderPermission(errCtx, b, p))
+			return nil
+		})
+	}
+
+	if err := errGrp.Wait(); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+func checkFolderPermission(ctx context.Context, b bundle.ReadOnlyBundle, folderPath string) diag.Diagnostics {
+	var diags diag.Diagnostics
+	w := b.WorkspaceClient().Workspace
+	obj, err := w.GetStatusByPath(ctx, folderPath)
```

Comment:
Are we sure these paths already exist when this validator is executed?

If not, we could check the permissions on containing folders seeing as those permissions will be inherited.

It also means we cannot do an equivalence check but rather have to make sure every permission on the workspace folder is included in the bundle permissions. The reverse, that all bundle permissions are also reflected on the actual permissions won't be true.